snap/snap.go

```diff
@@ -6,79 +6,174 @@ import (
 	"slices"
 
 	"github.com/go-spatial/geom"
+	"github.com/pdok/texel/intgeom"
 	"github.com/pdok/texel/processing"
+	"github.com/pdok/texel/tms20"
 	"github.com/umpc/go-sortedmap"
+	"golang.org/x/exp/constraints"
+)
+
+const (
+	internalPixelResolution = 16
 )
 
 // ToPointCloud snaps polygons' points to a tile's internal pixel grid
 // and adds points to lines to prevent intersections.
-func ToPointCloud(source processing.Source, target processing.Target, tileMatrix TileMatrix) {
-	processing.ProcessFeatures(source, target, func(p geom.Polygon) *geom.Polygon {
-		return snapPolygon(&p, tileMatrix)
+// targets is indexed by Tile Matrix ID
+func ToPointCloud(source processing.Source, targets map[int]processing.Target, tileMatrixSet *tms20.TileMatrixSet) {
+	processing.ProcessFeatures(source, targets, func(p geom.Polygon, tileMatrixIDs []int) map[int]*geom.Polygon {
+		return snapPolygon(&p, tileMatrixSet, tileMatrixIDs)
 	})
 }
 
-func snapPolygon(polygon *geom.Polygon, tileMatrix TileMatrix) *geom.Polygon {
-	ix := NewPointIndexFromTileMatrix(tileMatrix)
+func newPointIndexFromTileMatrixSet(tileMatrixSet *tms20.TileMatrixSet, deepestTileMatrixID int) *PointIndex {
```

Comment:
minor puntje: maar ik zou newPointIndexFromTileMatrixSet en tileMatrixIDsByLevels onder snapPolygon zetten. Zodat de file logischer van top-to-bottom te lezen is qua calls.